engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeConversionNode.java

```diff
@@ -172,10 +172,10 @@ Object doWarning(
       }
     }
     arguments[thatArgumentPosition] = that.getValue();
-    ArrayRope<Warning> warnings = that.getReassignedWarnings(this);
+    ArrayRope<Warning> warnings = that.getReassignedWarningsAsRope(this);
```

Comment:
At this point, when storing the warnings as sets, what is the point of getting these warnings as a rope?

Won't a simple array suffice? I don't see any reason to keep using `ArrayRope`s in the warning system at this moment.

Reply:
Not really. Gathering warnings means that we don't know the target size of the array/collection. So then a lot of methods that create/append collections/arrays would have to be behind truffle boundary as they are blacklisted. `ArrayRope<Warning>` is not an ideal data structure, yes. Eventually I would like to get rid of it completely but it would be a bigger change.
You also have to remember that this is added in generated code e.g. `ReadArrayElementMethodGen`.

Reply:
Oh, so we are appending the warnings using the ArrayRope and only at the end merging them into a Set? Ok I think I missed that part. Makes sense now.